sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala

```diff
@@ -82,6 +83,48 @@ class UDFSuite extends QueryTest {
     assert(ctx.sql("SELECT strLenScala('test', 1)").head().getInt(0) === 5)
   }
 
+  test("UDF in a WHERE") {
+    testData.sqlContext.udf.register("oneArgFilter", (n:Int) => { n > 80 })
+
+    val result =
+      testData.sqlContext.sql("SELECT * FROM testData WHERE oneArgFilter(key)")
+    assert(result.count() === 20)
+  }
+
+  test("UDF in a HAVING") {
+    testData.sqlContext.udf.register("havingFilter", (n:Long) => { n > 5 })
+
+    val result =
+      testData.sqlContext.sql("SELECT g, SUM(v) as s FROM groupData GROUP BY g HAVING havingFilter(s)")
```

Comment:
This line exceeds 100 characters and the last test failure is due to this. @spirom , Could you add proper indentation here?